src/Computer/GlideComputerAirData.cpp

```diff
@@ -189,14 +191,14 @@ GlideComputerAirData::CurrentThermal(const MoreData &basic,
                                      const CirclingInfo &circling,
                                      OneClimbInfo &current_thermal)
 {
-  if (circling.climb_start_time.IsDefined()) {
-    current_thermal.start_time = circling.climb_start_time;
-    current_thermal.end_time = basic.time;
-    current_thermal.gain =
-      basic.TE_altitude - circling.climb_start_altitude_te;
-    current_thermal.CalculateAll();
-  } else
-    current_thermal.Clear();
+  current_thermal.start_time = circling.climb_start_time;
+  current_thermal.end_time = basic.time;
+  double tmp_gain = basic.TE_altitude - circling.climb_start_altitude_te;
+  if ( tmp_gain < current_thermal.gain ) {
+    current_thermal.loss += tmp_gain - current_thermal.gain;
+  }
+  current_thermal.gain = tmp_gain;
+  current_thermal.CalculateAll();
```

Comment:
I don't understand this change.

Reply:
CurrentThermal() was never ever called with undefined climb_start_time time stamp. This was a sleeping bug and the Clear never got called. Whereas the Clear was not necessary until the addition of the "loss". The loss gets calculated incremental, the gain just as simple difference, no need for variable initialization.

Reply:
If you believe this is a bug, it should be fixed (and documented) in a separate commit.
The way you did this is hide it in an unrelated commit. That's bad.

Reply:
It is at least my best guess. Have not been keen enough to create a PR based on a best guess and w/o fixing an effective failure mode. 
Call it now the needed change for the additional loss attribute.

Reply:
If you're only guessing, then I'll refrain from merging this PR. I don't like merging code if the author is only guessing.

Reply:
I "guess" I dont understand you. My statement was that I refuse to push a so called fix on vague guess. What do you want to see here. Pushing a fix based on a gues .. not acting on a guess? Are you just messing up my words for fun or for not having the time reading?
The current implementation needs a zeroed start value, the current TC gain implementation does not. The straight forward coded change you see here.
Please advice clearly.

Reply:
I'm not messing with anything, I was just asking about a code change, and you said it fixed a bug, then you said it's just a guess. That confuses me.
If you think there's a bug, we should fix that bug first, before we discuss merging the feature this PR is about. But at least it should be a separate commit, before the commits adding this new feature, and the commit message should explain the bug being fixed. That's what I want to see (in the presence of a bug, one I don't know).

Reply:
So, I do not see a bug, nor a missbehavior of the software. I do see  a if - else, where the else never gets executed. For the intention of this I can only speculate. 
1) It could be that this "if" just was a safty measure in case CurrentThermal() would have been called with undefined TC start time stamp. all fine
2) It could have been that the intention/expectation was to really have the TC struct initialized when not in circling mode. Whereas, there is no need with the current feature set.
Useless to speculate, I easily can slip the change in a fix and a feature addition. Will force push the change and keep it in one PR. But we will have a commit commet that will be describing the change.

Reply:
If you don't know what this `if` is about and whether it's useful, then please don't submit a PR removing it. And, even more importantly: don't hide such a change in an unrelated commit.

Maybe it's really not useful, but maybe it is. We don't know. But let's not blindly hack the code without knowing what we're doing.